### DISCUSSION TITLE
Recording API FileInfo always writes tmp file first and requires the stopRecording to write the file

### DISCUSSION THREAD

**Discussion by p4xx07** [OvenMediaLabs/OvenMediaEngine#1281] [Help] [answered]
**Describe the bug**
In my request for recording I have as filepath "filePath" : "/path/to/file.ts"

Oven Media Engine will generate a tmp file named like "tmp_faisdfaildsf" with no extension. For what I'm doing I need the extension, I want to play from a frontend this file. Is there a way to directly have the file.ts

**To Reproduce**
1. Set Server.xml as follows   

 ```xml
<FILE>
    <RootPath>/mnt</RootPath>
<FilePath>/${VirtualHost}/${Application}/${Stream}/${StartTime:YYYYMMDDhhmmss}_${EndTime:YYYYMMDDhhmmss}.ts</FilePath>
<InfoPath>/${VirtualHost}/${Application}/${Stream}.xml</InfoPath>
</FILE>
```
3. With Encoder OBS ( the streaming is working fine )

**Expected behavior**
Record file to filename.extension and not write tmp_asodifasd to then move it to filename.extension after stopRecord

**Comment by getroot** [comment]
Until the recording ends, the ts file cannot be used because OME is writing. With split recording, you can do what you want.
Alternatively, you can use the LLHLS Dump function to record and use live VoD in near real time.
https://airensoft.gitbook.io/ovenmediaengine/streaming/low-latency-hls#dump

**Reply by p4xx07** [reply]
unfortunately this solution wouldn't work for what I'm doing. I want to use this file on a webplayer and I want to seek at different times.
If I did the interval split I would have to serve N files to the frontend and it would be very hard to reproduce the video with different pieces.

Is there anything else I could try?

Is there like a function that can force a split when I ask it to?

**Reply by getroot** [reply]
So what you want to do is play a VoD in the middle of a live recording. Is that right? Then use the LLHLS Dump function I mentioned above.

**Reply by p4xx07** [reply]
I will try thanks. I'm just wondering if the LLHLS dump will contain every part of the recording, or if it only contains part of the recording. We'll see

**Reply by p4xx07** [reply]
Ok so I've just tried. The LLHLS stream works fine. The issue is that I'm not able to seek backwards to the beginning of the recording. This doesn't work for my use case

**Reply by getroot** [reply]
VoDs created with LLHLS Dump play well in many environments. The seek you want works, of course. 

Are you sure you played the dumped playlist (m3u8) in OvenPlayer? OME is not a web server, so for this you will need a file based web server such as nginx. I think you are playing OME's LLHLS live stream. Since it is live, it will continue to delete past content unless you use the live rewind feature.

To solve your problem, I think you should first tell me how you did it, rather than just saying no.

**Reply by p4xx07** [reply]
I didn't use the OvenPlayer, I used as you said the LLHLS live stream. I see, thank you very much for your help.

**Reply by getroot** [reply, marked as answer]
If you want to go back to the beginning while watching a live broadcast like Youtube, use the Live Rewind (DVR) function. If you set the `MaxDuration `to a very long time, you can go back several hours even during a live broadcast.

https://airensoft.gitbook.io/ovenmediaengine/streaming/low-latency-hls#live-rewind

**Reply by p4xx07** [reply]
I'm going to try this. Thanks

**Reply by p4xx07** [reply]
Just to update you @getroot this solution seems to be great for my use case. I can easily serve the llhls and then work on the recording later. Thanks for the suggestion.
